app/services/uphold/models/transaction.rb

```diff
@@ -28,9 +28,28 @@ def initialize(params = {})
       def all(id:)
         Rails.logger.info("Connection #{@uphold_connection.id} is missing uphold_access_parameters") and return if @uphold_connection.uphold_access_parameters.blank?
 
-        response = get(PATH.expand(id: id), {}, authorization(@uphold_connection))
+        index = 1
+        batch_size = 50
 
-        parse_response(response).map { |a| Transaction.new(a) }
+        transactions = []
+
+        loop do
+          start = (batch_size * index) - batch_size
+          ending = start + batch_size
+
+          response = get(PATH.expand(id: id), {}, authorization(@uphold_connection), { "Range" => "items=#{start}-#{ending}" })
```

Comment:
Did you check to make sure this is exclusive of ending, i.e. it seems to query from 0 to 50 and then 50 to 100.

Reply:
Yes, per the Uphold API documentation -
```
The endpoints that support pagination returns a Content-Range header. For instance, if you make a request with Range: items=0-4 header the response will contain the following header: Content-Range: 0-4/* where * will be the total number of items that this endpoint can return.```